src/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,196 @@
+import React = require('react');
+import {EventHandler, ReactElement, SyntheticEvent} from "react";
+import {observable, action} from 'mobx';
+import {observer} from 'mobx-react';
+const style = require('./radio-group.st.css').default;
+
+export interface RadioGroupProps {
+    children?: any;
+    dataSource?: string[];
+    onChange: (e: string) => void;
+    disabled?: boolean;
+    location?: "right" | "left";
+    name?: string;
+}
+
+let counter = 0;
+
+export interface RadioState {
+    checked: boolean;
+}
+
+@observer
+export class RadioGroup extends React.Component<Partial<RadioGroupProps>, {}> {
+    name: string;
+    checkedArray: Array<RadioState>;
+
+    static defaultProps = {
+        dataSource: [],
+        location: 'right',
+        disabled: false
+    };
+
+    constructor(props: RadioGroupProps) {
+        super(props);
+        this.checkedArray = [];
+        let flag = true;
+        if (this.props.children) {
+            for (let i = 0; i < this.props.children.length; i++) {
+                this.checkedArray.push(observable({checked: flag ? this.props.children[i].props.checked: false}));
+                if (this.props.children[i].props.checked) {
+                    flag = false;
+                }
+            }
+        } else if(this.props.dataSource) {
+            for (let i = 0; i < this.props.dataSource.length; i++) {
+                this.checkedArray.push(observable({checked: false}));
+            }
+        }
+        this.name = this.props.name ? this.props.name : 'name_' + counter++;
+    }
+
+    childrenOnClick(index: number) {
+        return (e: string) => {
+            this.checkedArray.forEach((value) => {
+                value.checked = false;
+            });
+            this.checkedArray[index].checked = true;
+            if (this.props.onChange) {
+                this.props.onChange(e);
+            }
+        };
+    }
+
+    createChildren(dataArray: any, isDataSource: boolean) {
```

Comment:
Currently this is not easy to understand. I think it would be better to handle dataSource and children in two separate methods. And if want to reuse the props value in both, you can also extract it into a function.